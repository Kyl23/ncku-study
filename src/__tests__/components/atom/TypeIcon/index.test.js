import { render } from '@testing-library/react';
import TypeIcon from '~/src/components/atom/TypeIcon';

describe('test for atom - TypeIcon', () => {
    const themes = ['轉系', '輔系', '雙主修'];
    it(`render with first specify theme must render correctly`, () => {
        const { container } = render(<TypeIcon theme={themes[0]} />);

        expect(container).toMatchSnapshot();
    });

    it(`render with second specify theme must render correctly`, () => {
        const { container } = render(<TypeIcon theme={themes[1]} />);

        expect(container).toMatchSnapshot();
    });

    it(`render with third specify theme must render correctly`, () => {
        const { container } = render(<TypeIcon theme={themes[2]} />);

        expect(container).toMatchSnapshot();
    });
});
