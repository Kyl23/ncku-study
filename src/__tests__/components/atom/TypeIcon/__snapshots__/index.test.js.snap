// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for atom - TypeIcon render with first specify theme must render correctly 1`] = `
<div>
  <div
    class="css-xn8qaq"
  />
</div>
`;

exports[`test for atom - TypeIcon render with second specify theme must render correctly 1`] = `
<div>
  <div
    class="css-bhn2km"
  />
</div>
`;

exports[`test for atom - TypeIcon render with third specify theme must render correctly 1`] = `
<div>
  <div
    class="css-1t9gcyy"
  />
</div>
`;
